pkg/console/controllers/util/util.go

```diff
@@ -52,11 +54,36 @@ func SliceContains(s []string, value string) bool {
 	return false
 }
 
-func IsSupportedVersion(productVersion string) bool {
+func IsSupportedManagedClusterVersion(productVersion string) bool {
 	version, err := semver.Parse(productVersion)
 	if err != nil {
 		klog.V(4).Infof("unable to parse %q version", productVersion)
 		return false
 	}
 	return version.Compare(semver.MustParse("4.0.0")) == 1
 }
+
+func IsSupportedMangedCluster(managedCluster clusterv1.ManagedCluster) (bool, bool) {
+	isValidProduct := false
+	isValidVersion := false
+	for _, claim := range managedCluster.Status.ClusterClaims {
+		if claim.Name == api.ManagedClusterClaimProductAnnotation && SliceContains(api.SupportedClusterProducts, claim.Value) {
+			isValidProduct = true
+			continue
+		}
+		if claim.Name == api.ManagedClusterClaimVersionAnnotation && IsSupportedManagedClusterVersion(claim.Value) {
+			isValidVersion = true
+			continue
+		}
+	}
+	return isValidProduct, isValidVersion
+}
+
+func IsConditionMet(conditions []metav1.Condition, conditionType string, conditionStatus metav1.ConditionStatus) bool {
```

Comment:
WE already have `getClusterVersionCondition` if `/pkg/console/controllers/upgradenotification/controller.go`. Lets update it's params so we can use `IsConditionMet` instead

Reply:
I found an equivalent helper in the k8s apimachinery package: https://github.com/kubernetes/apimachinery/blob/6ecd1c8969025104b032f6a7f14ae93afc50e107/pkg/api/meta/conditions.go#L95